### PR TITLE
[Inquiries] Conditionally enable submit button.

### PR DESCRIPTION
**Pull request by alloy**
conditionally-enable-submit-button

### CODE REVIEW

File: src/apps/loyalty/containers/inquiries/index.tsx [hunk 1]
```diff
@@ -91,16 +95,14 @@ export class Inquiries extends React.Component<RelayProps, State> {
   }
 
   onArtworkSelected(conversationId: string, selected: boolean) {
-    const selectedArtworks = this.state.selected_artworks
-    selectedArtworks[conversationId] = selected
-
     this.setState({
-      selected_artworks: selectedArtworks,
+      selected_artworks: Object.assign({}, this.state.selected_artworks, { [conversationId]: selected }),
```
Comment (alloy) [inline src/apps/loyalty/containers/inquiries/index.tsx]: Just making sure that in-place editing of `this.state.selected_artworks` doesn’t lead to side-effects (now or in the future).

File: src/apps/loyalty/containers/login/__tests__/__snapshots__/index.tsx.snap [hunk 2]
```diff
@@ -64,7 +64,7 @@ exports[`login renders the snapshot 1`] = `
       className="eTVvmy eYaNZQ"
       href="/twitter"
       icon={
-        <Styled(Icon)
+        <Styled(Component)
```
Comment (alloy) [inline src/apps/loyalty/containers/login/__tests__/__snapshots__/index.tsx.snap]: Hmm, this fails on CI. I thought it was weird that these snapshots didn’t match on my machine anyways, so must be something else. I’ll revert this so the PR is mergable and look into why it fails on my machine afterwards.

File: src/apps/loyalty/containers/3w_thank_you/index.tsx [hunk 3]
```diff
@@ -51,3 +55,5 @@ export default props => (
     </ButtonWrapper>
   </Container>
 )
+
+export default ThreeWThankYou
```
Comment (mzikherman) [inline src/apps/loyalty/containers/3w_thank_you/index.tsx]: Got it.

File: src/apps/loyalty/containers/inquiries/index.tsx (hunk 1 above)
Comment (mzikherman) [inline src/apps/loyalty/containers/inquiries/index.tsx]: 👍

File: src/apps/loyalty/containers/inquiries/index.tsx [hunk 4]
```diff
@@ -148,6 +150,11 @@ export class Inquiries extends React.Component<RelayProps, State> {
     alert("Sorry, there was an error with your submission, please try again")
   }
 
+  enableSubmitButton() {
+    const { selected_artworks, self_reported_purchases } = this.state
+    return !!self_reported_purchases || Object.keys(selected_artworks).some(id => selected_artworks[id])
```
Comment (mzikherman) [inline src/apps/loyalty/containers/inquiries/index.tsx]: Nice! I love `some`

**Comment by mzikherman** [comment]
LGTM! 🎉

File: src/apps/loyalty/containers/inquiries/__tests__/index.tsx [hunk 5]
```diff
@@ -50,11 +51,48 @@ describe("inquiries", () => {
...
-    const aTags = ReactTestUtils.scryRenderedDOMComponentsWithTag(inquiries, "a")
+    const inquiries = TestUtils.renderIntoDocument(<Inquiries {...inquiryProps}/>) as Inquiries
+    const aTags = TestUtils.scryRenderedDOMComponentsWithTag(inquiries, "a") as HTMLAnchorElement[]
     const artistLink = aTags.find(tag => tag.href === "/percy-z" && tag.textContent === "Percy Z")
     expect(artistLink).toBeTruthy()
     const noConvoLink = aTags.find(tag => tag.href === "/percy-z-without-conversation")
     expect(noConvoLink).toBeUndefined()
   })
+
+  describe("concerning the submit button", () => {
```
Comment (mzikherman) [inline src/apps/loyalty/containers/inquiries/__tests__/index.tsx]: nice tests

**Pull request merged**